### PR TITLE
Renaming, template handlers and registry

### PR DESCRIPTION
**Pull request by afdw**
Fixes https://github.com/BuildCraft/BuildCraft/issues/3177.

### CODE REVIEW

File: api/buildcraft/api/enums/EnumHandlerPriority.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2017 SpaceToad and the BuildCraft team
+ * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not
+ * distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/
+ */
```
Comment (AlexIIL) [inline api/buildcraft/api/enums/EnumHandlerPriority.java]: Wrong license - API is licensed as MIT, not MPLv2

File: api/buildcraft/api/transport/IStripesHandlerBlock.java [hunk 2]
```diff
@@ -6,13 +6,16 @@
...
      * @param player
      * @param activator
      * @return True if this broke a block, false otherwise (note that this handler MUST NOT return false if it has
      *         changed the world in any way) */
-    boolean handle(World world, BlockPos pos, EnumFacing direction, EntityPlayer player, IStripesActivator activator);
+    boolean handle(World world,
+                   BlockPos pos,
+                   EnumFacing direction,
+                   EntityPlayer player,
+                   IStripesActivator activator);
```
Comment (AlexIIL) [inline api/buildcraft/api/transport/IStripesHandlerBlock.java]: Please don't reformat like this.

File: api/buildcraft/api/core/IFakePlayerProvider.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import net.minecraftforge.common.util.FakePlayer;
+
+public interface IFakePlayerProvider {
+    /**
+     * Returns the generic buildcraft fake player. Note that you shouldn't use this anymore, as you should store the
+     * UUID of the real player who created the block or entity that calls this.
+     */
+    @Deprecated
+    FakePlayer getBuildCraftPlayer(WorldServer world);
```
Comment (AlexIIL) [inline api/buildcraft/api/core/IFakePlayerProvider.java:22]: Don't add this deprecate method to the API - we'll only remove it later in source code, and I'd rather not keep support for it around in the api.

File: api/buildcraft/api/core/BuildCraftAPI.java [hunk 4]
```diff
@@ -18,9 +18,17 @@
...
+import buildcraft.api.transport.pipe.IPipeRegistry;
+import buildcraft.api.transport.pluggable.IPluggableRegistry;
 
-    public static IBCFakePlayer fakePlayerProvider;
+public final class BuildCraftAPI {
+    public static IFakePlayerProvider fakePlayerProvider;
+    public static IPipeRegistry pipeRegistry;
+    public static IPluggableRegistry pluggableRegistry;
+    public static IStripesRegistry stripeRegistry;
+    public static ITemplateRegistry templateRegistry;
```
Comment (AlexIIL) [inline api/buildcraft/api/core/BuildCraftAPI.java]: It's common for mods to only include the packages they need in the built jars, so we need to remove inter-package dependcies. This needs doing for the whole api (I'll do it alongsde my pass of the main code cleanup when everything is closer to completion) so for now please move this back into PipeApi. (You can leave templateRegistry there or add a TemplateApi class to put it in)

File: api/buildcraft/api/enums/EnumHandlerPriority.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+public enum EnumHandlerPriority implements Comparable<EnumHandlerPriority> {
+    HIGHEST,
+    HIGH,
+    NORMAL,
+    LOW,
+    LOWEST;
+
+    public static final EnumHandlerPriority[] VALUES = values();
+}
```
Comment (AlexIIL) [inline api/buildcraft/api/enums/EnumHandlerPriority.java]: Please leave this class where it was, the enum package also needs api cleanup (as mentioned above)

**Review by AlexIIL — CHANGES_REQUESTED** [review]

File: api/buildcraft/api/core/IFakePlayerProvider.java (hunk 3 above)
Comment (afdw) [reply api/buildcraft/api/core/IFakePlayerProvider.java:22]: I just moved it from old place, not add.

File: api/buildcraft/api/enums/EnumHandlerPriority.java (hunk 5 above)
Comment (afdw) [reply api/buildcraft/api/enums/EnumHandlerPriority.java]: This class was inner class of one of registry, where should I move it?

Comment (afdw) [reply api/buildcraft/api/enums/EnumHandlerPriority.java]: I'll move it `core` package.

File: api/buildcraft/api/core/IFakePlayerProvider.java (hunk 3 above)
Comment (AlexIIL) [reply api/buildcraft/api/core/IFakePlayerProvider.java:22]: Sorry, I got confused by the method addition (which is fine BTW)

**Review by AlexIIL — APPROVED** [review]

**Comment by AlexIIL** [comment]
Looks good, thanks :)

**Pull request merged**